Modules/_ssl.c

```diff
@@ -1410,14 +1410,61 @@ _get_peer_alt_names (X509 *certificate) {
                 PyTuple_SET_ITEM(t, 1, v);
                 break;
 
+            case GEN_IPADD:
+                /* OpenSSL < 3.0.0 adds a trailing \n to IPv6. 3.0.0 removed
+                 * the trailing newline. Remove it in all versions
+                 */
+                t = PyTuple_New(2);
```

Comment:
It might be slightly more convenient to do `PyTuple_Pack` after marshaling the values.

Reply:
The other cases use the same approach. Let's not mix styles.